### PR TITLE
Fix successful Apache Druid task submissions reported as failed

### PR DESCRIPTION
**Pull request by listik**
Apache Druid returns status code `202 Approved` on successful MSQ requests, but the hook is only checking for a `200` status code.

This means MSQ ingestions are reported as failed by default, despite being successfully executed by the Druid service.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by potiuk** [comment]
Could you please add a unit test covering it?

File: airflow/providers/apache/druid/hooks/druid.py [hunk 1]
```diff
@@ -107,9 +107,9 @@ def submit_indexing_job(
         req_index = requests.post(url, data=json_index_spec, headers=self.header, auth=self.get_auth())
 
         code = req_index.status_code
-        if code != 200:
+        if code not in [200, 202]:
```
Comment (dirrao) [inline airflow/providers/apache/druid/hooks/druid.py]: isn't it good idea to check for 2xx instead of specific codes?

**Comment by listik** [comment]
> Could you please add a unit test covering it?

Hey @potiuk, I thought [this unit test](https://github.com/apache/airflow/blob/7a814f1403752d42d309f79010d9214083eda762/tests/providers/apache/druid/hooks/test_druid.py#L81-L95) was already covering the `submit_indexing_job()` hook method?

**Comment by potiuk** [comment]
> Hey @potiuk, I thought [this unit test](https://github.com/apache/airflow/blob/7a814f1403752d42d309f79010d9214083eda762/tests/providers/apache/druid/hooks/test_druid.py#L81-L95) was already covering the submit_indexing_job() hook method?

Apparently it does not test what it should then. If you make a change fixing a problem and there is no test that fails before and get fixed after the change, it means that the error case has not been tested.

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Thanks for adding the test!

**Comment by listik** [comment]
> Thanks for adding the test!

Thank you for the feedback and patience!

**Pull request merged**